src/exec/handle_redirect.c

```diff
@@ -70,28 +72,29 @@ static void	redirect_std_fd(t_redirect_type type, int fd)
 	}
 }
 
-bool	handle_redirect(t_list *redirects, t_ctx *ctx)
+bool	handle_redirect(t_list *redirect, t_ctx *ctx)
```

Comment:
suggestion
```c
bool	handle_redirect(t_list *redirect, t_ctx *ctx)
{
	int			fd;
	const char	*expanded_filepath;

	expanded_filepath = expand_filepath(redirect, ctx);
	if (expanded_filepath == NULL)
	{
		print_error(get_redirect_file_or_delim(redirect), "ambiguous redirect");
		set_file_or_delim(redirect, NULL);
		return (false);
	}
	set_file_or_delim(redirect, expanded_filepath);
	fd = open_redirect_file(redirect->content, ctx);
	if (fd == -1)
	{
		if (get_redirect_type(redirect) != REDIRECT_HEREDOC)
			print_error(get_redirect_file_or_delim(redirect), strerror(errno));
		return (false);
	}
	redirect_std_fd(get_redirect_type(redirect), fd);
	close(fd);
	return (true);
}
```

Reply:
It is better than current one.
But it is need to add the below line.
free((char *)get_redirect_file_or_delim(redirect));